File: ckgame-java/src/main/java/com/ckgame/politics/Claim.java
package com.ckgame.politics;

import com.ckgame.core.Constants;

/**
 * 角色对头衔或领地的宣称。
 */
public final class Claim {
    public final int claimant;
    public final int title;
    public final int county;
    public boolean pressed = false;
    public int strength = 50;

    public Claim(int claimant, int title, Integer county, int strength) {
        this.claimant = claimant;
        this.title = title;
        this.county = county != null ? county : Constants.NONE_ID;
        this.strength = strength;
    }
}

File: ckgame-ts/src/politics/Claim.ts
import { NONE_ID } from '../core/Constants.js';

/**
 * A character's claim on a title or county.
 */
export class Claim {
  public pressed: boolean = false;
  public strength: number;

  constructor(
    public readonly claimant: number,
    public readonly title: number,
    public readonly county: number | null,
    strength?: number,
  ) {
    this.strength = strength ?? 50;
    // Normalize null county to NONE_ID for internal use
    if (this.county === null || this.county === undefined) {
      (this as { county: number }).county = NONE_ID;
    }
  }
}
